dynamallow/local.py

```diff
@@ -0,0 +1,80 @@
+import os
+import random
+import socket
+import tarfile
+import tempfile
+
+import subprocess
+
+try:
+    from urllib import urlretrieve
+except ImportError:
+    from urllib.request import urlretrieve
+
+
+class DynamoLocal(object):
+    """
+    Spins up a local dynamo instance. This should ONLY be used for testing!! You must call
+    ``dynamo_local.shutdown()`` when you are done with the instance.
+    """
+    def __init__(self, dynamo_dir, port=None, log=None):
```

Comment:
It's an anti-pattern pass loggers around like this.

Every module that needs to use a logger should create a logger at the top of the module using the pattern:

    log = logging.getLogger(__name__)

This allows consumers to fetch and alter logging settings using the python module namespace.  An example is that if you want debug logging in your app but find the vendored urllib3 module inside of requests too chatty you can:

    logging.getLogger("requests.packages.urllib3").setLevel(logging.INFO)

Or to change all of requests:

    logging.getLogger("requests").setLevel(logging.INFO)